master/buildbot/db/logs.py

```diff
@@ -328,7 +328,8 @@ def thdcompressLog(conn):
             # calculate how many bytes we saved
             q = sa.select([sa.func.sum(sa.func.length(tbl.c.content))])
             q = q.where(tbl.c.logid == logid)
-            newsize = conn.execute(q).fetchone()[0]
+            # if fetchone()[0] is None, then set it to 0
+            newsize = conn.execute(q).fetchone()[0] or 0
```

Comment:
I can't find a reason why fetchone()[0] will return None.

I would prefer to fix the rootcause rather than hiding the problem under the carpet.
Do you have any more context of what is the log that is being compressed? is it empty or anyhow special?